packages/mui-material/src/Slider/SliderValueLabel.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_generateUtilityClasses as generateUtilityClasses } from '@mui/utils';
+import { sliderUnstyledClasses } from '@mui/base/SliderUnstyled';
+import { SliderValueLabelProps } from './SliderValueLabel.types';
+
+export const sliderClasses = {
+  ...sliderUnstyledClasses,
+  ...generateUtilityClasses('MuiSlider', [
+    'colorPrimary',
+    'colorSecondary',
+    'thumbColorPrimary',
+    'thumbColorSecondary',
+    'sizeSmall',
+    'thumbSizeSmall',
+  ]),
+};
+
+const useValueLabelClasses = (open: SliderValueLabelProps['open']) => {
+  const utilityClasses = {
+    offset: clsx({
+      [sliderClasses.valueLabelOpen]: open,
+    }),
+    circle: sliderClasses.valueLabelCircle,
+    label: sliderClasses.valueLabelLabel,
+  };
+
+  return utilityClasses;
+};
+
+/**
+ * @ignore - internal component.
+ */
+export default function SliderValueLabel(props: SliderValueLabelProps) {
+  const { children, className, value, open } = props;
+  const classes = useValueLabelClasses(open);
+
+  if (!children) {
+    return null;
+  }
```

Comment:
This differs with the implementation with MUI Base, can you elaborate why it was added?

Reply:
`children` property is typed optional in `SliderValueLabelProps`, so TypeScript complains that it's possibly null.

Implementation in MUI Base is with Javascript.

Reply:
Thanks for the explanation 👍